### PR TITLE
[81] Fix "Widget is Dispose" Error when closing diagram

### PR DESCRIPTION
**Pull request by scosta-obeo**
Bug: https://github.com/eclipse-sirius/sirius-desktop/issues/81

### CODE REVIEW

File: plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/editor/tabbar/TabbarToolBarManager.java [hunk 1]
```diff
@@ -60,8 +57,7 @@ public class TabbarToolBarManager extends ToolBarManager {
     public TabbarToolBarManager(ToolBar toolBar, IWorkbenchPart part) {
         super(toolBar);
         this.tabbarPart = part;
-        this.onResize = event -> updateGradientBackground();
-        getControl().addListener(SWT.Resize, onResize);
+        getControl().addListener(SWT.Resize, event -> updateGradientBackground());
```
Comment (lredor) [inline plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/editor/tabbar/TabbarToolBarManager.java:60]: Add a comment to explain that the listener will be automatically removed. This allows to clarify why there is no "getControl().removeListener" later.

Comment (lredor) [inline plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/editor/tabbar/TabbarToolBarManager.java:60]:
```suggestion
        //This listener will be automatically "removed/clean" during the release of the tabbar.
        getControl().addListener(SWT.Resize, event -> updateGradientBackground());
```

File: plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/editor/tabbar/TabbarToolBarManager.java [hunk 2]
```diff
@@ -214,7 +210,6 @@ private Color createColor(int color1, int color2, int ratio, Display display) {
 
     @Override
     public void dispose() {
```
Comment (lredor) [inline plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/editor/tabbar/TabbarToolBarManager.java:212]: As explained orally, I prefer that you restore the order of methods calls (`super.dispose()` in first), as the change order has no link with the issue #81.

If you want, you can add another "cleanup" commit affter.

**Review by lredor — CHANGES_REQUESTED** [review]

**Comment by lredor** [comment]
Please also add a comment in the commit description to explain that:

This problem occurs since the last commit for #81, ie commit [1] f9b33566.

[1] https://github.com/eclipse-sirius/sirius-desktop/commit/f9b33566602c30877c907d8fa4d904feece411fb

File: plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/editor/tabbar/TabbarToolBarManager.java (hunk 2 above)
Comment (scosta-obeo) [reply plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/editor/tabbar/TabbarToolBarManager.java:213]: Done

File: plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/editor/tabbar/TabbarToolBarManager.java (hunk 1 above)
Comment (scosta-obeo) [reply plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/editor/tabbar/TabbarToolBarManager.java:61]: Done

Comment (scosta-obeo) [reply plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/editor/tabbar/TabbarToolBarManager.java:61]: Done

**Review by lredor — APPROVED** [review]

**Comment by lredor** [comment]
> Please also add a comment in the commit description to explain that:
> 
> This problem occurs since the last commit for #81, ie commit [1] [f9b3356](https://github.com/eclipse-sirius/sirius-desktop/commit/f9b33566602c30877c907d8fa4d904feece411fb).
> 
> [1] [f9b3356](https://github.com/eclipse-sirius/sirius-desktop/commit/f9b33566602c30877c907d8fa4d904feece411fb)

@scosta-obeo This comment has not been considered.

**Pull request merged**